extractor.go

```diff
@@ -154,7 +101,7 @@ func (ex *ExtractorService) listen(w io.Writer, blockSub, txsSub types.Subscript
 		}
 		// we need to drain all
 
-		if err := indexBlock(w, sync, eventData); err != nil {
+		if err := indexBlock(ex.writer, sync, eventData); err != nil {
```

Comment:
fail on nil? can it be nil at all?

Reply:
You mean writer being nil or the error? The writer is always going to be set. As far as aborting on error - we should introduce some kind of a keyword, like `DMLOG ABORT <.....error message?>` to indicate that something went terribly wrong and we cant continue. Afaik the indexing service can't abort the node unless we let it crash and burn.

Reply:
taking about people f** up config files. If it really always will be set and no won't crash on this, just disreagard this comment :)

Reply:
Yea we're defaulting to STDOUT